### DISCUSSION TITLE
Live preview for PHP

### DISCUSSION THREAD

**Discussion by joncode21** [phcode-dev/phoenix#1755] [General]
Hi 

Can anybody help in terms of "Live Preview" for my PHP files

When I try to do a custom "Live Preview Settings" with a port of 80 or 8080 followed by my own selected folder,the "live preview" would display a random generated port number along with a file path of its own on Google Chrome search bar

I have also read the docs in Phoneix Code website, but it does not help with my situation.

**Comment by abose** [comment]
Can you provide a screenshot of the live preview settings dialogue?
The URL you see in chrome is ransom as phcode injects its own instrumentation to the webpage to debug the page better. So the URL may not reflect the actual URL in the address bar.

**Comment by joncode21** [comment]
![phoenix](https://github.com/user-attachments/assets/07b50e3b-5c45-4e31-af1b-38e269f40f1c)

**Reply by abose** [reply]
This should work, what is the url being loaded in the live prevew?

**Reply by charlypa** [reply]
@joncode21 

WOuld you share your server used for your development?

**Reply by joncode21** [reply]
> This should work, what is the url being loaded in the live prevew?


Link:
http://localhost:53474/StatichbbPzBkm/live-preview-navigator.html?initialURL=http%3A%2F%2Flocalhost%3A53474%2FStatichbbPzBkm%2Fphoenix-splash%2Fno-preview.html%3FjsonInput%3D%25257B%252522heading%252522%25253A%252522Custom%252520Server%252520Cannot%252520Serve%252520This%252520file%252522%25252C%252522details%252522%25253A%252522Live%252520preview%252520settings%252520is%252520configured%252520to%252520only%252520serve%252520files%252520from%252520folder%252520%2527%25252Fschool%2527%252522%25257D%26phcodeLivePreview%3Dtrue&livePreviewCommURL=ws%3A%2F%2Flocalhost%3A53474%2FPreviewCommQtL0Ct0r&isLoggingEnabled=false

**Reply by joncode21** [reply]
> @joncode21
> 
> WOuld you share your server used for your development?

what do u mean?

**Reply by abose** [reply]
Ah, it means that Phoenix will only show live preview for files in the /school folder

Ie, /school/page.html will be mapped to localhost:80/page.html

Web pages outside of /school folder will never be shown.

**Reply by abose** [reply]
So , if just leave the folder as `/` if you meant that all files in the project are served by the server and not just contents in the `/school` folder

**Reply by joncode21** [reply]
> So , if just leave the folder as `/` if you meant that all files in the project are served by the server and not just contents in the `/school` folder

OK, but the issue is that I am unable to display anything from the "/school" folder.So what do I do? Do I need to specify which file I wan to view in the "Live preview settings"?

**Comment by charlypa** [comment]
@joncode21 
Which server are you using for  live preview your PHP files.

**Reply by joncode21** [reply]
> @joncode21 Which server are you using for live preview your PHP files.

XAMPP Control Panel

**Comment by charlypa** [comment]
@joncode21 
it works for me
![image](https://github.com/user-attachments/assets/aac97d64-5deb-4ee1-9ffe-cb63a208188f)
Following is live preview settings
![image](https://github.com/user-attachments/assets/a124dea4-96d7-4520-a959-3a48b61b46f0)

you may need to update XAMPP conf Aswell to point to your project folder
```
DocumentRoot "C:/Users/charl/repo/yourdir"
<Directory "C:/Users/charl/repo/yourdir">
    #
    # Possible values for the Options directive are "None", "All",
    # or any combination of:
    #   Indexes Includes FollowSymLinks SymLinksifOwnerMatch ExecCGI MultiViews
    #
    # Note that "MultiViews" must be named *explicitly* --- "Options All"
    # doesn't give it to you.
    #
    Options Indexes FollowSymLinks Includes ExecCGI

    #
    # AllowOverride controls what directives may be placed in .htaccess files.
    # It can be "All", "None", or any combination of the keywords:
    #   AllowOverride FileInfo AuthConfig Limit
    #
    AllowOverride All

    #
    # Controls who can get stuff from this server.
    #
    Require all granted
</Directory>
```

After doing this you have to restart server from XAMPP panel

![image](https://github.com/user-attachments/assets/b214734e-6a47-4a04-8248-e4fa4fb0613c)

Your `httpd.conf` located in `C:\xampp\apache\conf`

**Reply by joncode21** [reply]
ok it works now!

**Reply by joncode21** [reply]
now there another issue, the "Server supports hot reload" button in the "Live Preview Settings" does not  change automatically unless I save the file and do a live preview again

**Reply by bal5** [reply]
@charlypa
I did what you wrote above, I get a blank page, it is written on the blue bar above that 'getting preview from your custom server', but stays blank. When I click 'open in Firefox', it opens without problem. What can be missing?

**Comment by joncode21** [comment]
To anyone who wonders how it works


1. the XAMPP config file  you will have to specify the "directory" and "document root" to the folder that you want it to run live
        a. how to go:  C:\xampp\apache\conf\httpd.conf ,
        b.next open it in notepad 
        c.go to the "Document Root" section
        d. Specify the folder in both  "directory" and "document root" that you wan the editor to display in live preview
        

2.In the "live preview settings", the"serving folder in project " , just put a "/" would do (Do not include the inverted commas)

3. and volia the thing works.

**Comment by charlypa** [comment]
@joncode21 
Created a document to setup live preview for php
https://docs.phcode.dev/docs/Features/php-live-preview/
Hope this helps

**Reply by joncode21** [reply]
now there another issue, the "Server supports hot reload" button in the "Live Preview Settings" does not change automatically unless I save the file and do a live preview again

**Reply by charlypa** [reply]
@joncode21 
We are aware of it created a ticket to track this
https://github.com/phcode-dev/phoenix/issues/1768

**Comment by joncode21** [comment]
Hi just wondering, if I do this custom setting, am I able to use it for phpmyAdmin or Do i have to revert back to the default setting?

**Reply by charlypa** [reply]
@joncode21 
You should be able to use phmyAdmin as is ideally it should not affect you. phpmyAdmin is for the database. LivePreveiw has nothing to do with a database.

**Reply by joncode21** [reply]
yup i tried, phpmyadmin can be accessed,  i realise is my code that has error.

**Comment by joncode21** [comment]
Hello ,the instructions no longer work at all, in terms of "live preview settings" and the "httpd.conf" file, may I know how do I go about live preview now??

**Reply by joncode21** [reply]
ok i tried now, but now it requires me to specify the particular child folder in the "httpd.conf" file.
In the past ,I was able to just specify the parent folder - "School" and it worked
